src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -414,23 +438,23 @@ private bool TrySteal(out IThreadPoolWorkItem obj, ref bool missedSteal, int mil
         internal class QueueSegment
         {
             // Holds a segment of the queue.  Enqueues/Dequeues start at element 0, and work their way up.
-            internal readonly IThreadPoolWorkItem[] nodes;
+            internal readonly PaddedWorkItem[] nodes;
```

Comment:
Same question here: why are you padding this?  What contention are you trying to prevent? Two threads operating on the same segment are already going to be contending on indexes, so it's not clear to me how padding each node element and the indexes and the Next field really helps.  You have numbers to show that it does?  This is again adding significant overhead in memory.  Previously on 64-bit each of these nodes arrays would have been ~2K (8 bytes * 256 elements). Now they're each ~16K (64 bytes * 256 elements).